diff --git a/libs/circle/addon/fatfs/diskio.cpp b/libs/circle/addon/fatfs/diskio.cpp
index 086bfdbd..0d21a311 100644
--- a/libs/circle/addon/fatfs/diskio.cpp
+++ b/libs/circle/addon/fatfs/diskio.cpp
@@ -33,11 +33,26 @@ static const char *s_pVolumeName[FF_VOLUMES] =
 	"umsd2",
 	"umsd3",
 	"ufd1",
-	"nvme1"
+	"nvme1",
+	"unused"
 };
 
 static CDevice *s_pVolume[FF_VOLUMES] = {0};
 
+#if FF_MULTI_PARTITION
+PARTITION VolToPart[FF_VOLUMES] =
+{
+	{0, 1},		// SYS:  SD card partition 1
+	{0, 2},		// USER: SD card partition 2
+	{0, 1},		// SD:   legacy alias for SYS:
+	{2, 0},		// USB:
+	{3, 0},		// USB2:
+	{4, 0},		// USB3:
+	{5, 0},		// FD:
+	{6, 0}		// NVME:
+};
+#endif
+
 static u8 *s_pBuffer = 0;
 static unsigned s_nBufferSize = 0;
 
diff --git a/libs/circle/addon/fatfs/ffconf.h b/libs/circle/addon/fatfs/ffconf.h
index 9011a9a8..96b14708 100644
--- a/libs/circle/addon/fatfs/ffconf.h
+++ b/libs/circle/addon/fatfs/ffconf.h
@@ -178,12 +178,12 @@
 / Drive/Volume Configurations
 /---------------------------------------------------------------------------*/
 
-#define FF_VOLUMES		7
+#define FF_VOLUMES		8
 /* Number of volumes (logical drives) to be used. (1-10) */
 
 
 #define FF_STR_VOLUME_ID	1
-#define FF_VOLUME_STRS		"SD","SD2","USB","USB2","USB3","FD","NVME"
+#define FF_VOLUME_STRS		"SYS","USER","SD","USB","USB2","USB3","FD","NVME"
 /* FF_STR_VOLUME_ID switches support for volume ID in arbitrary strings.
 /  When FF_STR_VOLUME_ID is set to 1 or 2, arbitrary strings can be used as drive
 /  number in the path name. FF_VOLUME_STRS defines the volume ID strings for each
@@ -196,7 +196,7 @@
 */
 
 
-#define FF_MULTI_PARTITION	0
+#define FF_MULTI_PARTITION	1
 /* This option switches support for multiple volumes on the physical drive.
 /  By default (0), each logical drive number is bound to the same physical drive
 /  number and only an FAT volume found on the physical drive will be mounted.
