--- a/libs/circle/lib/usb/usbdevicefactory.cpp
+++ b/libs/circle/lib/usb/usbdevicefactory.cpp
@@ -130,7 +130,8 @@
 		pResult = new CUSBGamePadPS4Device (pParent);
 	}
 	else if (   pName->Compare ("ven45e-28e") == 0
-		 || pName->Compare ("ven45e-28f") == 0)
+		 || pName->Compare ("ven45e-28f") == 0
+		 || pName->Compare ("ven2dc8-3106") == 0)	// 8BitDo XInput receiver
 	{
 		pResult = new CUSBGamePadXbox360Device (pParent);
 	}
